src/core/conversion/dataIndexer.ts

```diff
@@ -0,0 +1,157 @@
+        this.coordinates = Array.from({ length: Object.keys(targetMap).length });
+        this.variableOrder = this.getVariableOrder(completeMap, targetMap);
+        for (let targetIndex = 0; targetIndex < Object.keys(targetMap).length; targetIndex++) {
+            const variableIndex: number = this.variableOrder[targetIndex];
+            const targetVariableCode = Object.keys(targetMap)[targetIndex];
+            const variable: string[] = completeMap[targetVariableCode];
+            if (!variable) throw new Error("Provided variable code can not be found from the metadata");
+            this.coordinates[variableIndex] = Array.from({ length: targetMap[targetVariableCode].length });
+            for (let mapIndex = 0; mapIndex < targetMap[targetVariableCode].length; mapIndex++) {
+                for (let valueIndex = 0; valueIndex < variable.length; valueIndex++) {
+                    if (variable[valueIndex] === targetMap[targetVariableCode][mapIndex]) {
+                        this.coordinates[variableIndex][mapIndex] = valueIndex;
+                    }
+                }
+            }
+        }
+        this.indices = Array.from({ length: Object.keys(completeMap).length }, () => 0);
+        this.lastIndices = Object.keys(targetMap).map(v => targetMap[v].length - 1);
+        this.lastCoordinateIndex = this.coordinates.length - 1;
+        this.reverseCumulativeProducts = this.generateRCP(variableSizes);
+        this.dataLength = Object.keys(targetMap).map(v => targetMap[v].length).reduce((acc, val) => acc * val, 1);
+        this.dataIndex = 0;
+        this.setCurrentIndex();
+    }
+
+    public getViewSeries(viewPrecision: number | null): IDataSeries[] {
+        const viewSeries: IDataSeries[] = [];
+        do {
+            const { rowNameGroup, dataCell } = this.calculateIndex(viewPrecision);
+            const group: IDataSeries | undefined = viewSeries.find(s => s.rowNameGroup.every(g => rowNameGroup.includes(g) && rowNameGroup.every(g => s.rowNameGroup.includes(g))));
+            if (group) {
+                group.series.push(dataCell);
+            }
+            else {
+                viewSeries.push({
+                    rowNameGroup: rowNameGroup,
+                    series: [dataCell]
+                })
+            }
+        } while (this.next());
+
+        return viewSeries;
+    }
+
+    next(): boolean {
+        for (let i = this.lastCoordinateIndex; i >= 0; i--) {
+            const variableIndex: number = this.variableOrder[i];
+            if (this.indices[variableIndex] < this.lastIndices[i]) {
+                this.indices[variableIndex]++;
+                this.setCurrentIndex();
+                return true;
+            }
+            else {
+                this.indices[variableIndex] = 0;
+            }
+        }
+        return false;
+    }
+
+    calculateIndex(viewPrecision: number | null): { rowNameGroup: TMultiLanguageString[], dataCell: IDataCell } {
```

Comment:
We should use the precision information from the content dimension value here instead of the view precision.